util/formats/newfmt/types.go

```diff
@@ -0,0 +1,217 @@
+	Description string                   `json:"description"`
+	TopoSVG     string                   `json:"topo_svg"` // Base64-encodes SVG
+	Cards       []SysinfoGpuCard         `json:"cards"`
+	Software    []SysinfoSoftwareVersion `json:"software"`
+}
+
+type SysinfoCoreModel struct {
+	Index    uint64 `json:"index"`
+	Physical uint64 `json:"physical"`
+	Model    string `json:"model"`
+}
+
+type SysinfoGpuCard struct {
+	Index         uint64 `json:"index"`
+	UUID          string `json:"uuid"`
+	Address       string `json:"address"`
+	Manufacturer  string `json:"manufacturer"`
+	Model         string `json:"model"`
+	Architecture  string `json:"architecture"`
+	Driver        string `json:"driver"`
+	Firmware      string `json:"firmware"`
+	Memory        uint64 `json:"memory"`
+	PowerLimit    uint64 `json:"power_limit"`
+	MaxPowerLimit uint64 `json:"max_power_limit"`
+	MinPowerLimit uint64 `json:"min_power_limit"`
+	MaxCEClock    uint64 `json:"max_ce_clock"`
+	MaxMemClock   uint64 `json:"max_memory_clock"`
+}
+
+type SysinfoSoftwareVersion struct {
+	Key     string `json:"key"`
+	Name    string `json:"name"`
+	Version string `json:"version"`
+}
+
+// JSON representation can be read with ConsumeJSONSamples().
+
+type SampleEnvelope struct {
+	Data   *SampleData    `json:"data"`
+	Errors []ErrorObject  `json:"errors"`
+	Meta   MetadataObject `json:"meta"`
+}
+
+type SampleData struct {
+	Type       string           `json:"type"`
+	Attributes SampleAttributes `json:"attributes"`
+}
+
+type SampleAttributes struct {
+	Time    string        `json:"time"`
+	Cluster string        `json:"cluster"`
+	Node    string        `json:"node"`
+	Attrs   []KVPair      `json:"attrs"`
+	System  SampleSystem  `json:"system"`
+	Jobs    []SampleJob   `json:"jobs"`
+	Errors  []ErrorObject `json:"errors"`
+}
+
+type SampleSystem struct {
+	Cpus       []SampleCpu `json:"cpus"`
+	Gpus       []SampleGpu `json:"gpus"`
+	UsedMemory uint64      `json:"used"`
+}
+
+type SampleCpu = uint64
+
+type SampleGpu struct {
+	Index      uint64 `json:"index"`
+	UUID       string `json:"uuid"`
+	Bad        bool   `json:"bad"`
+	Fan        uint64 `json:"fan"`
+	Mode       string `json:"mode"`
+	Perf       int64  `json:"perf"`
+	Memory     uint64 `json:"memory"`
+	CEUtil     uint64 `json:"ce_util"`
+	MemUtil    uint64 `json:"memory_util"`
+	Temp       int64  `json:"temp"`
+	Power      uint64 `json:"power"`
+	PowerLimit uint64 `json:"power_limit"`
+	CEClock    uint64 `json:"ce_clock"`
+	MemClock   uint64 `json:"memory_clock"`
+}
+
+type SampleJob struct {
+	Job       uint64          `json:"job"`
+	User      string          `json:"user"`
+	Processes []SampleProcess `json:"processes"`
+}
+
+type SampleProcess struct {
+	Resident  uint64             `json:"resident"`
+	Virtual   uint64             `json:"virtual"`
+	Cmd       string             `json:"cmd"`
+	Pid       uint64             `json:"pid"`
+	ParentPid uint64             `json:"ppid"`
+	CpuAvg    float64            `json:"cpu_avg"`
```

Comment:
These four fields - two here, two in gpu data - need not be floating point, they could be uint.  They are percentage points, not fractions of 1.

Reply:
I will not do this now.  They have been float forever, and they can change to integer later without affecting any clients that consume untyped data a la JSON (typed data - maybe not so much).